thirdparty/build_thirdparty.sh

```diff
@@ -47,18 +48,31 @@ export PATH=$PREFIX/bin:$PATH
 
 # build snappy
 if [ -n "$F_ALL" -o -n "$F_SNAPPY" ]; then
-  cd $SNAPPY_DIR
+  cd $TP_DIR/$SNAPPY_BASEDIR
   ./configure --with-pic --prefix=$PREFIX
   make -j$PARALLEL install
 fi
 
 # build lz4
 if [ -n "$F_ALL" -o -n "$F_LZ4" ]; then
-  cd $LZ4_DIR
+  cd $TP_DIR/$LZ4_BASEDIR/cmake_unofficial
   CFLAGS=-fPIC cmake -DCMAKE_INSTALL_PREFIX:PATH=$PREFIX $LZ4_DIR
   make -j$PARALLEL install
 fi
 
+# build thrift
+if [ -n "$F_ALL" -o -n "$F_THRIFT" ]; then
+  if [ "$(uname)" == "Darwin" ]; then
+      echo "thrift compilation under OSX is not currenlty supported."
```

Comment:
Yeah, I wasn't quite sure how to handle this.  I meant for the exit 1 to indicate that it couldn't install thift on OSX and if it were asked to do so then it would be an error.  I suppose I could make it just a warning that doesn't fail the script.  Does that sound better?